datafusion/core/src/datasource/listing/table.rs

```diff
@@ -843,10 +843,22 @@ impl TableProvider for ListingTable {
             });
         // TODO (https://github.com/apache/datafusion/issues/11600) remove downcast_ref from here?
         let session_state = state.as_any().downcast_ref::<SessionState>().unwrap();
-        let (mut partitioned_file_lists, statistics) = self
-            .list_files_for_scan(session_state, &partition_filters, limit)
+        // TODO avoid these clones when possible.
+        let session_state_captured = session_state.clone();
```

Comment:
There are quite a few more call sites like this that would need to be annotated for the correct runtime. However I didn't bother to do it unless we could figure out how to make the rest of the API reasonable (aka `run_cpu` ,etc)